uhabits-core/src/jvmTest/java/org/isoron/uhabits/core/preferences/PreferencesTest.kt

```diff
@@ -162,6 +165,31 @@ class PreferencesTest : BaseUnitTest() {
         assertFalse(prefs.showCompleted)
     }
 
+    @Test
+    @Throws(Exception::class)
+    fun testActiveNotifications() {
+        repeat(5) { habitList.add(fixtures.createEmptyHabit()) }
+
+        // Initially no active notifications
+        assertThat(prefs.getActiveNotifications(habitList), equalTo(HashMap()))
+
+        // Example map of active notifications
+        val a = HashMap<Habit, NotificationTray.NotificationData>()
+        for (i in listOf(0, 1, 3)) {
+            val habit = habitList.getByPosition(i)
+            val data = NotificationTray.NotificationData(Timestamp(10000L * i), 200000L * i)
+            a[habit] = data
+        }
+
+        // Persist and retrieve active notifications
+        prefs.setActiveNotifications(a)
+        val b = prefs.getActiveNotifications(habitList)
+
+        // Assert that persisted and retrieved maps are teh same
+        assertThat(a.keys, equalTo(b.keys))
+        a.forEach { e -> assertThat(b[e.key], equalTo(e.value)) }
```

Comment:
This is a manual workaround for the missing matcher/library function I was looking for.